readthedocs/proxito/exceptions.py

```diff
@@ -0,0 +1,100 @@
+from django.http import Http404
+from django.utils.translation import gettext_lazy as _
+
+
+class ContextualizedHttp404(Http404):
+
+    """
+    Base class for contextualized HTTP 404 handling.
+
+    Subclasses may define their own template name,
+    HTTP status and object that was not found.
+
+    The contextualized exception is handled by the project's 404 view.
+    """
+
+    template_name = "errors/404/base.html"
+    not_found_subject = _("page")
+
+    def __init__(self, **kwargs):
+        """
+        Constructor that all subclasses should call.
+
+        :param kwargs: all kwargs are added as page context for rendering the 404 template
+        :param http_status: 404 view should respect this and set the HTTP status.
+        :param path_not_found: Inform the template and 404 view about a different path from
+                               request.path
+        """
+        self.http_status = kwargs.pop("http_status", 404)
+        self.path_not_found = kwargs.pop("path_not_found", None)
+        self.kwargs = kwargs
+
+    def get_context(self):
+        c = {
+            "not_found_subject": self.not_found_subject,
+            "path_not_found": self.path_not_found,
+        }
+        c.update(self.kwargs)
+        return c
+
+
+class DomainDNSHttp404(ContextualizedHttp404):
```

Comment:
It's easy to introduce bugs here, making a typo while passing the kwargs since they don't have any validation. I'd prefer if we are explicit about what the template needs.

Reply:
I've added positional arguments where they are required :+1: